.github/workflows/docs.yml

```diff
@@ -0,0 +1,80 @@
+name: Docs CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Git repositories
+      - name: Checkout Source
+        uses: actions/checkout@v2
+        with:
+          path: PandABlocks-webcontrol
+          # require history to get back to last tag for version number of branches
+          fetch-depth: 0
+
+      # System dependencies (required for build)
+      - name: Install system dependencies
+        run: |
+          sudo pip install sphinx sphinx-rtd-theme --upgrade docutils==0.16
+
+      - name: Checkout annotypes
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/annotypes
+          path: annotypes
+
+      - name: Checkout pymalcolm
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/pymalcolm
+          path: pymalcolm
+
+      - name: Checkout malcolmjs
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/malcolmjs
+          path: malcolmjs
+
+      - name: Checkout PandABlocks-rootfs
+        uses: actions/checkout@v2
+        with:
+          repository: PandABlocks/PandABlocks-rootfs
+          path: PandABlocks-rootfs
+
+      # CONFIG files
+      - name: Generate CONFIG files
+        run: |
+          sudo bash PandABlocks-webcontrol/.github/scripts/config-file-webcontrol.sh
+
+      # Build
+      - name: Build Docs
+        run: |
+          cd PandABlocks-webcontrol
+          make docs
+
+      - uses: rishabhgupta/split-by@v1
+        id: split
+        with:
+          string: ${{ github.ref }}
+          split-by: /
+
+      # Publish docs
+      - name: Move to versioned directory
+        # e.g. master or 0.1.2
+        run: mv PandABlocks-webcontrol/build/templates "PandABlocks-webcontrol/.github/pages/${{ steps.split.outputs._2 }}"
```

Comment:
Found a better way to do this:
https://github.com/dls-controls/scanspec/blob/master/.github/workflows/docs.yml#L41-L43